mpas_analysis/shared/io/namelist_streams_interface.py

```diff
@@ -0,0 +1,91 @@
+#!/usr/bin/python
+"""
+    Module of classes / routines to manipulate fortran namelist and streams
+    files.
+
+    Phillip Wolfram
+    LANL
+    04/24/2014
+"""
+
+from subprocess import check_output, call
+from lxml import etree
+
+class NameList:
+    """
+    Class for fortran manipulation of namelist files, provides
+    read and write functionality
+    """
+
+    # constructor
+    def __init__(self, fname, readonly=True):
+        # set readonly bit
+        self.readonly = readonly
+        # input file name
+        self.fname = fname
+
+    def read(self, name):
+        # shell return value
+        return_val = check_output(['awk', '/'+name+'/{printf $3}', self.fname])
+        return_val.rstrip('\n')
+        try:
+            return_val = float(return_val)
+        except:
+            return_val = return_val.strip('"').strip("'")
+        return return_val
+
+    def write(self, name, value):
+        assert not self.readonly, 'Cannot write to namelist file %s because readonly=True'%(self.fname)
+        # make sure value is converted to a string
+        if (type("") != type(value)):
+            value = str(value)
+        read_val = check_output(['awk', '/'+name+'/{printf $3}', self.fname])
+        read_val = read_val.strip('"')
+        call('cp '+self.fname+' '+self.fname+'.backup',shell=True)
+
+        #print ['sed', '/'+name+'/s/'+read_val+'/'+value+'/ ',  self.fname+'.backup'+' > '+self.fname]
+        command = "awk '/"+ name+'/ && !done {sub("'+read_val+'","'+ value+"\"); done=1}; 1' < "+self.fname+'.backup' + " > "+self.fname
+        #print command
+        result = call(command,shell=True)
+
+class XMLList:
+    """
+    Class to read in streams configuration file, provdies
+    read and write functionality
+    """
+
+    def __init__(self, fname, backup=True, readonly=True):
+        self.fname = fname
+        self.readonly = readonly
+        self.xmlfile = etree.parse(fname)
+        self.root = self.xmlfile.getroot()
+        if backup:
```

Comment:
If you do decide to keep write/modify functionality, I think backup=True only makes sense if readonly=False.